src/scrape/scrape_job_test.go

```diff
@@ -200,6 +233,40 @@ func TestPopulateK8s(t *testing.T) {
 	}
 }
 
+func TestRestartCountDeltaValues(t *testing.T) {
+	intgr, err := integration.New("test", "test", integration.InMemoryStore())
+	assert.NoError(t, err)
+	intgr.Clear()
+
+	expectedRestartCountDeltas := []float64{0, 3, 0, 1}
+
+	grouper := &testGrouper{
+		ContainerValuesInTime: map[string][]interface{}{
+			"restartCount": {0, 3, 3, 4},
+		},
+	}
+	testJob := NewScrapeJob("test", grouper, kubeletSpecs)
+
+	k8sVersion := &version.Info{GitVersion: "v1.15.42"}
+	// Populate data several times to check expected deltas
+	for i := 0; i < len(expectedRestartCountDeltas); i++ {
+		err = testJob.Populate(intgr, "test-cluster", logutil.Debug, k8sVersion)
+		require.IsType(t, err, data.PopulateResult{})
+		assert.Empty(t, err.(data.PopulateResult).Errors)
+		time.Sleep(time.Second)
+	}
+
+	for _, entity := range intgr.Entities {
+		if entity.Metadata.Name == "newrelic-infra" {
+			lenExpectedDeltas := len(expectedRestartCountDeltas)
+			require.Equal(t, lenExpectedDeltas, len(entity.Metrics))
+			for i := 0; i < lenExpectedDeltas; i++ {
+				assert.Equal(t, expectedRestartCountDeltas[i], entity.Metrics[i].Metrics["restartCountDelta"])
```

Comment:
I do not completely get how (why?) this is working. `entity.Metrics` is a `MetricSet`. Why do we have `n` `MetricSet`s, where `n` is the number of times we called `Populate`? Is this intended behavior of the code?